supporting_scripts/analysis-of-endpoint-connections/src/main/typeScript/Preprocessor.ts

```diff
@@ -0,0 +1,320 @@
+     * It then constructs a `ChildClass` object containing information about the current class, such as its name,
+     * an empty map for member variables, and an empty array for parent method calls.
+     * This `ChildClass` object is added to the `PREPROCESSING_RESULTS`, which track the relationships between superclasses
+     * and their child classes. If the superclass is already present in the map, the current class is added to its
+     * list of child classes. Otherwise, a new entry for the superclass is created in the map.
+     * Finally, the method calls `identifySuperConstructorCalls` to process any calls to the superclass constructor
+     * within the current class.
+     *
+     * @param classDeclaration - The class declaration node to process.
+     */
+    identifySuperClass(classDeclaration: TSESTree.ClassDeclaration) {
+        let superClassName: string;
+        let superClassPath: string;
+        if (classDeclaration.superClass && classDeclaration.superClass.type === 'Identifier') {
+            superClassName = classDeclaration.superClass.name;
+            superClassPath = this.identifyImportedClassByName(superClassName);
+            const childClassName = classDeclaration.id?.name ?? '';
+            const childClass = { superClass: superClassName, name: childClassName, memberVariables: this.memberVariables, parentMethodCalls: [] };
+
+            const superConstructorCall = this.identifySuperConstructorCalls(classDeclaration, superClassName);
+
+            if (Preprocessor.PREPROCESSING_RESULTS.has(superClassName)) {
+                Preprocessor.PREPROCESSING_RESULTS.get(superClassName)?.childClasses.push(childClass);
+                Preprocessor.PREPROCESSING_RESULTS.get(superClassName)?.superConstructorCalls.concat(... superConstructorCall)
+            } else {
+                Preprocessor.PREPROCESSING_RESULTS.set(superClassName, { name: superClassName, path: superClassPath, childClasses: [childClass], superConstructorCalls: superConstructorCall});
+            }
+        }
+    }
+
+    /**
+     * Identifies calls to the superclass constructor within a class declaration.
+     * This method iterates over the body of the class to find the constructor method.
+     * Within the constructor, it looks for call expressions to `super` and evaluates their arguments.
+     * The evaluated arguments are collected and returned as an array of objects, each containing an `arguments` array.
+     *
+     * @param classDeclaration - The class declaration node to process.
+     * @param superClassName - The name of the superclass to identify constructor calls for.
+     * @returns An array of objects, each containing an `arguments` array with the evaluated values of the super constructor call arguments.
+     */
+    identifySuperConstructorCalls(classDeclaration: TSESTree.ClassDeclaration, superClassName: string) {
+        const result: {arguments: string[]}[]= [];
```

Comment:
The formatting looks a little off here
```suggestion
        const result: { arguments: string[] }[] = [];
```